test/integration-tests/IntegrationTests.Http/HttpTests.cs

```diff
@@ -0,0 +1,86 @@
+// <copyright file="HttpTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Linq;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using IntegrationTests.Helpers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace IntegrationTests.Http
+{
+    public class HttpTests : TestHelper
+    {
+        private const string ServiceName = "Samples.Http";
+
+        public HttpTests(ITestOutputHelper output)
+            : base("Http", output)
+        {
+            SetEnvironmentVariable("OTEL_SERVICE_NAME", ServiceName);
+            SetEnvironmentVariable("OTEL_DOTNET_AUTO_ENABLED_INSTRUMENTATIONS", "HttpClient,AspNet");
+        }
+
+        [Fact]
+        [Trait("Category", "EndToEnd")]
+        public void SubmitTraces()
+        {
+            var agentPort = TcpPortProvider.GetOpenPort();
+            using var agent = new MockZipkinCollector(Output, agentPort);
+
+            const int expectedSpanCount = 3;
+
+            using var processResult = RunSampleAndWaitForExit(agent.Port, enableStartupHook: true);
+            Assert.True(processResult.ExitCode >= 0, $"Process exited with code {processResult.ExitCode} and exception: {processResult.StandardError}");
+            var spans = agent.WaitForSpans(expectedSpanCount, 3000);
+
+            using (new AssertionScope())
+            {
+                spans.Count.Should().Be(expectedSpanCount);
+
+                // ASP.NET Core auto-instrumentation is generating spans
+                var httpClientSpan = spans.FirstOrDefault(span => span.Name.Equals("HTTP GET"));
+                var httpServerSpan = spans.FirstOrDefault(span => span.Name.Equals("/test"));
+                var manualSpan = spans.FirstOrDefault(span => span.Name.Equals("manual span"));
+
+                httpClientSpan.Should().NotBeNull();
```

Comment:
consider grouping assertion by [`AssertionScope`s](https://fluentassertions.com/introduction#assertion-scopes).
If more than one assertion fails it produces message with all issues instead of only first one.

Reply:
There is a single assertion scope used for all of them but one

Reply:
Sorry, I do not know how I can missed that. :)